### PR TITLE
CCv0: Refactor signature verification PoC changes

### PR DESCRIPTION
**Pull request by Megan-Wright**
Add signature verification required artefacts and copy them to the guest
This replaces code that was previously in the CCv0 script in `kata-containers`
The files added are: 
- `signatures.tar` - a tar archive containing the signatures of `quay.io/kata-containers/confidential-containers:signed` and `quay.io/kata-containers/confidential-containers:other_signed`
- `public.gpg` - the public GPG key, paired to the private key pair that was used to sign `quay.io/kata-containers/confidential-containers:signed`
- `quay_policy.json` - a container policy file that allows insecure access to all repos except `quay.io/kata-containers`, in which it enforced signatures by the above key

Fixes: #4810
Depends-on: github.com/kata-containers/kata-containers#4354

Signed-off-by: Megan Wright <megan.wright@ibm.com>
Co-authored-by: Georgina Kinge <Georgina.Kinge@ibm.com>

### CODE REVIEW

File: integration/confidential/lib.sh [hunk 1]
```diff
@@ -174,3 +175,17 @@ load_runtime_config_path() {
...
+	agent_config_files=("${SHARED_FIXTURES_DIR}/public.gpg" "${SHARED_FIXTURES_DIR}/quay_policy.json" "${SHARED_FIXTURES_DIR}/signatures.tar" "${SHARED_FIXTURES_DIR}/quay.io.yaml")
+	cp_to_guest_img "${rootfs_quay_verification_directory}" "${agent_config_files[@]:0:2}"
+		cp_to_guest_img "${rootfs_quay_verification_directory}" "${agent_config_files[@]:0:2}"
+	if [ ! -d "${SHARED_FIXTURES_DIR}/signatures" ]; then
+		mkdir "${SHARED_FIXTURES_DIR}/signatures"
+	fi
+	sudo tar -zvxf "${agent_config_files[2]}" -C "${SHARED_FIXTURES_DIR}/signatures"
+	cp_to_guest_img "${rootfs_quay_verification_directory}" "${SHARED_FIXTURES_DIR}/signatures"
+	cp_to_guest_img "${container_registries_dir}" "${agent_config_files[3]}"
+}
```
[the remark this answers is not in the document]
Comment (Megan-Wright) [reply integration/confidential/lib.sh]: @stevenhorsman yep that makes complete sense - I didn't think of it as the support for the copying over directories was one of the last things we added.... updated the code now :)

**Review by stevenhorsman — APPROVED** [review]
LGTM - Thanks!

**Comment by stevenhorsman** [comment]
@wainersm - this looks good to me, if you have time to take a quick look as well, that would be fab!

**Review by GabyCT — APPROVED** [review]

**Review by wainersm — APPROVED** [review]
hi @Megan-Wright , it looks good to me! thanks!

**Pull request merged**